base/expr.jl

```diff
@@ -532,11 +532,12 @@ macro assume_effects(args...)
         elseif setting === :nothrow
             nothrow = true
         elseif setting === :terminates_globally
-            terminates_globally = true
+            terminates_globally = terminates_locally = true
```

Comment:
Wondering whether it would be better for debugging not to set this override and instead just also check for the global override also on the backedge.